lib/mbr.ml

```diff
@@ -296,3 +296,42 @@ let marshal (buf : Cstruct.t) t =
 
 let sizeof = sizeof_mbr
 let default_partition_start = 2048l
+
+let print_mbr_fields mbr =
+  Printf.printf "MBR fields:\n";
+  Printf.printf "  bootstrap_code: %s\n" mbr.bootstrap_code;
+  Printf.printf "  original_physical_drive: %d\n" mbr.original_physical_drive;
+  Printf.printf "  seconds: %d\n" mbr.seconds;
+  Printf.printf "  minutes: %d\n" mbr.minutes;
+  Printf.printf "  hours: %d\n" mbr.hours;
+  Printf.printf "  disk_signature: %ld\n" mbr.disk_signature;
```

Comment:
It would be great to print the disk signature as hexadecimal. This is what `fdisk -l test.img` does.

Reply:
for this, I was exploring the `Cstruct` library and found a useful function `to_hex_string`, however this function exist only in `v6.2.0` while the latest release of `Cstruct` on Opam is `v6.1.1`

The PR to publish the new release is still pending. 
https://github.com/ocaml/opam-repository/pull/23374

I am unsure if it's wise to wait or find some other way to convert the `Cstruct.t` value to a hex string.

Reply:
For `disk_signature` you can use `%lx` in printf.

Good question about `bootstrap_code`. There is a 'hex' library that can be used. That would introduce another dependency for the binary, but that is maybe alright. Another way forward could be to print "<boostrap code omitted>" since I think that's rarely what you're interested in. And then later when cstruct.6.2.0 is released we can add a flag to print the bootstrap code.

Reply:
I think the best way will be to print an empty string and update later when cstruct 6.2.0 is released.
I think adding another dependency (hex library) will be too much given that we are using it just for this purpose.

Reply:
:-) Thanks for the `%lx`. It's just what I needed.